src/user/mod.rs

```diff
@@ -3,34 +3,94 @@ use std::collections::HashMap;
 use conduit::{Request, Response};
 use conduit_cookie::{RequestSession};
 use conduit_router::RequestParams;
+use diesel::prelude::*;
+use diesel::pg::PgConnection;
 use pg::GenericConnection;
 use pg::rows::Row;
 use rand::{thread_rng, Rng};
 
-use {Model, Version};
 use app::RequestApp;
 use db::RequestTransaction;
+use {http, Model, Version};
 use krate::{Crate, EncodableCrate};
+use schema::users;
 use util::errors::NotFound;
 use util::{RequestUtils, CargoResult, internal, ChainError, human};
 use version::EncodableVersion;
-use http;
 
 pub use self::middleware::{Middleware, RequestUser};
 
 pub mod middleware;
 
 /// The model representing a row in the `users` database table.
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug, PartialEq, Eq, Queryable)]
 pub struct User {
     pub id: i32,
-    pub gh_login: String,
-    pub gh_id: i32,
-    pub name: Option<String>,
     pub email: Option<String>,
-    pub avatar: Option<String>,
     pub gh_access_token: String,
     pub api_token: String,
+    pub gh_login: String,
+    pub name: Option<String>,
+    pub avatar: Option<String>,
```

Comment:
Wait, `Queryable` cares about the order of the fields on `User` but it doesn't care that the field here is named `avatar` but the field in the db is named `gh_avatar`?